src/arrayMethodJoin.js

```diff
@@ -4,8 +4,27 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    if (this[0] === null) {
+      this[0] = '';
+    }
+
+    let result = '' + this[0];
+    const workSepar = '' + separator;
+
+    if (this.length === 0) {
+      return '';
+    }
+
+    for (let i = 1; i <= this.length - 1; i++) {
+      if (this[i] === null || this[i] === undefined) {
+        this[i] = '';
+      }
+
+      result += workSepar + this[i];
+    }
+
+    return result;
```

Comment:
for checklist everything is fine, but workSepar(its strange name for variable) maybe you wanted to say wordSeparator?

Reply:
good point